### PR TITLE
Eb/deeplink integration

### PR DESCRIPTION
**Pull request by ghost**
- Created `useDeeplinkUrl` to be able to listen deeplinks in React Native.
- Added sample deeplink integration for Android
- Added sample deeplink integration for iOS

### CODE REVIEW

**Review by yunusinitialcode — APPROVED** [review]

File: src/Hooks/useDeeplinkUrl.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  useEffect(() => {
+    // If the app is already open, the app is foregrounded and a Linking 'url' event is fired
+    Linking.addEventListener(EVENT_NAME, ({url}) => {
+      setDeeplinkUrl(url);
+    });
+
+    // If the app is not already open, it is opened and the url is passed in as the initialURL
+    const getUrlAsync = async () => {
+      const initialUrl = await Linking.getInitialURL();
```
Comment (yasirinitialcode) [inline src/Hooks/useDeeplinkUrl.ts:17]: This function returning same value all the time. Did you test it 
1) opened with deep link (closed app)
2) send a new deeplink event
conditions? 
first step trigger getInitialURL but second step trigger eventListener. I am not sure about URL's final value
~Nejdet

File: src/Hooks/useDeeplinkUrl.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import {useEffect, useState} from 'react';
+import {Linking} from 'react-native';
+
+export const useDeeplinkUrl = () => {
+  const [deeplinkUrl, setDeeplinkUrl] = useState<string | null>(null);
+
+  const EVENT_NAME = 'url';
+
+  useEffect(() => {
+    // If the app is already open, the app is foregrounded and a Linking 'url' event is fired
```
Comment (yasirinitialcode) [inline src/Hooks/useDeeplinkUrl.ts]: Can we express comments a bit clear, i did not quite comprehend what it does mean ?

File: src/Screens/PushInbox.tsx [hunk 3]
```diff
@@ -255,7 +255,7 @@ const PushInbox = () => {
         renderItem={({ item, index }) => {
           return getInboxItem(item, index);
         }}
-        keyExtractor={item => item.pushInstanceId}
+        keyExtractor={item => item.pushInstanceId ?? ""}
```
Comment (yasirinitialcode) [inline src/Screens/PushInbox.tsx]: Wouldn't `keyExtractor={(item, index) => item.pushInstanceId ?? index.toString()}` be more sensible.

**Review by yasirinitialcode — APPROVED** [review]

File: src/Hooks/useDeeplinkUrl.ts (hunk 2 above)
Comment (ghost) [reply src/Hooks/useDeeplinkUrl.ts]: I copied this comment from the official document 😅

**Pull request merged**